tfsdk/serve_test.go

```diff
@@ -1874,14 +1899,318 @@ func TestServerPlanResourceChange(t *testing.T) {
+					"interface": tftypes.NewValue(tftypes.String, "scsi"),
+				}),
+			}),
+			resource:                "test_three",
+			resourceType:            testServeResourceTypeThreeType,
+			expectedRequiresReplace: []*tftypes.AttributePath{tftypes.NewAttributePath().WithAttributeName("scratch_disk").WithAttributeName("interface")},
+		},
+		"three_attrplanmodifiers_nested_modifier": {
+			priorState: tftypes.NewValue(testServeResourceTypeThreeType, map[string]tftypes.Value{
+				"name": tftypes.NewValue(tftypes.String, "name1"),
+				"size": tftypes.NewValue(tftypes.Number, 3),
+				"scratch_disk": tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{
+					"id":        tftypes.String,
+					"interface": tftypes.String,
+				}}, map[string]tftypes.Value{
+					"id":        tftypes.NewValue(tftypes.String, "my-scr-disk"),
+					"interface": tftypes.NewValue(tftypes.String, "scsi"),
+				}),
+			}),
+			proposedNewState: tftypes.NewValue(testServeResourceTypeThreeType, map[string]tftypes.Value{
+				"name": tftypes.NewValue(tftypes.String, "name1"),
+				"size": tftypes.NewValue(tftypes.Number, 3),
+				"scratch_disk": tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{
+					"id":        tftypes.String,
+					"interface": tftypes.String,
+				}}, map[string]tftypes.Value{
+					"id":        tftypes.NewValue(tftypes.String, "TESTATTRTWO"),
+					"interface": tftypes.NewValue(tftypes.String, "scsi"),
+				}),
+			}),
+			config: tftypes.NewValue(testServeResourceTypeThreeType, map[string]tftypes.Value{
+				"name": tftypes.NewValue(tftypes.String, "name1"),
+				"size": tftypes.NewValue(tftypes.Number, 3),
+				"scratch_disk": tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{
+					"id":        tftypes.String,
+					"interface": tftypes.String,
+				}}, map[string]tftypes.Value{
+					"id":        tftypes.NewValue(tftypes.String, "TESTATTRTWO"),
+					"interface": tftypes.NewValue(tftypes.String, "scsi"),
+				}),
+			}),
+			expectedPlannedState: tftypes.NewValue(testServeResourceTypeThreeType, map[string]tftypes.Value{
+				"name": tftypes.NewValue(tftypes.String, "name1"),
+				"size": tftypes.NewValue(tftypes.Number, 3),
+				"scratch_disk": tftypes.NewValue(tftypes.Object{AttributeTypes: map[string]tftypes.Type{
+					"id":        tftypes.String,
+					"interface": tftypes.String,
+				}}, map[string]tftypes.Value{
+					"id":        tftypes.NewValue(tftypes.String, "MODIFIED_TWO"),
+					"interface": tftypes.NewValue(tftypes.String, "scsi"),
+				}),
+			}),
+			resource:                "test_three",
+			resourceType:            testServeResourceTypeThreeType,
+			expectedRequiresReplace: []*tftypes.AttributePath{tftypes.NewAttributePath().WithAttributeName("scratch_disk").WithAttributeName("interface")},
+		},
 	}
 
 	for name, tc := range tests {
 		name, tc := name, tc
 
 		t.Run(name, func(t *testing.T) {
-			t.Parallel()
```

Comment:
Hmm, why did this need to stop running in parallel? 🤔 Would love a comment here explaining why it can't run in parallel. That makes me mildly worried about race conditions.

Reply:
Thanks, I don't think this removal is needed any more since I got to the bottom of the race conditions. Will restore it.